tests/Functional/ResultCacheTest.php

```diff
@@ -402,6 +432,37 @@ public function testChangeCacheImplLegacy(): void
         self::assertCount(1, $secondCache->fetch('emptycachekey'));
     }
 
+    public function testChangeCacheImplLegacyWrapped(): void
+    {
+        $stmt = $this->connection->executeQuery(
+            'SELECT * FROM caching WHERE test_int > 500',
+            [],
+            [],
+            new QueryCacheProfile(10, 'emptycachekey')
+        );
+
+        $this->hydrateViaIteration($stmt, static function (Result $result) {
+            return $result->fetchAssociative();
+        });
+
+        $arrayAdapter = new ArrayAdapter();
+        $secondCache  = DoctrineProvider::wrap($arrayAdapter);
+
+        $stmt = $this->connection->executeQuery(
+            'SELECT * FROM caching WHERE test_int > 500',
+            [],
+            [],
+            new QueryCacheProfile(10, 'emptycachekey', $secondCache)
+        );
+
+        $this->hydrateViaIteration($stmt, static function (Result $result) {
+            return $result->fetchAssociative();
+        });
+
+        self::assertCount(2, $this->sqlLogger->queries, 'two hits');
```

Comment:
This test is a bit hard to understand, especially because of this comment, which I know you only copied, but still… if I am correct this hits are database hits, not cache hits. If that's true, let's make it clear.
```suggestion
        self::assertCount(2, $this->sqlLogger->queries, 'two database hits');
```

Reply:
I have adjusted all `assetCounts()` calls on `$this->sqlLogger->queries` and added meaningful messages.